google/cloud/spanner/connection_options.h

```diff
@@ -82,12 +86,24 @@ class ConnectionOptions {
     return *this;
   }
 
+  ConnectionOptions& add_user_agent_prefix(std::string prefix) {
```

Comment:
What do users use this for? Is it documented somewhere how a user could/should use the connection's user-agent prefix?

Reply:
The typical use-case is a library built on top of the spanner client (think TensorFlow, or something like [SOCI](https://github.com/SOCI/soci) or [sqlpp11](https://github.com/rbock/sqlpp11)), or a service built on top of our library (don't know, maybe Cloud Dataflow or something similar).

They would want to add their own user-agent prefix for the same reasons we do.